client/src/components/modal/component.tsx

```diff
@@ -0,0 +1,112 @@
+import { useRef } from 'react';
```

Comment:
Pretty much the same Modal from https://github.com/Vizzuality/front-end-scaffold, with a few tweaks to fit the project.